torch/fx/graph.py

```diff
@@ -269,12 +269,20 @@ def __len__(self):
         return self.graph._len
 
     def __iter__(self):
-        root, direction = self.graph._root, self.direction
-        cur = getattr(root, direction)
-        while cur is not root:
-            if not cur._erased:
-                yield cur
-            cur = getattr(cur, direction)
+        root = self.graph._root
+        if self.direction == "_next":
```

Comment:
Optimizations like this make me sad, I feel like these are things compiler should be able to do, but I guess it requires us to use a statically compiled language

Reply:
I'm writing graph.find_nodes(op, target) right now which will remove majority of the fx graph iteration, so hopefully that will be enough